src/town_gui.cpp

```diff
@@ -1249,16 +1249,10 @@ struct FoundTownWindow : Window {
 				break;
 
 			case WID_TF_MANY_RANDOM_TOWNS: {
-				Backup<bool> old_generating_world(_generating_world, true);
-				UpdateNearestTownForRoadTiles(true);
-				if (!GenerateTowns(this->town_layout)) {
-					ShowErrorMessage(GetEncodedString(STR_ERROR_CAN_T_GENERATE_TOWN), GetEncodedString(STR_ERROR_NO_SPACE_FOR_TOWN), WL_INFO);
-				}
-				UpdateNearestTownForRoadTiles(false);
-				old_generating_world.Restore();
+				std::string default_town_number = (_settings_game.difficulty.number_towns == (uint)CUSTOM_TOWN_NUMBER_DIFFICULTY) ? GetString(STR_JUST_INT, _settings_newgame.game_creation.custom_town_number) : std::to_string(Map::ScaleBySize(_num_initial_towns[_settings_game.difficulty.number_towns] + (Random() & 7)));
```

Comment:
* If we ever get a network-multi-designer scenario editor: `Random` must not be used in GUI code (it will cause desyncs). GUI code must use `InteractiveRandom`.
* Anyhow, in this case I think it's better to just skip the random part. I think it looks weird for the user, if there is a different number every time.
* We are currently trying to get rid of stream-based string formatting, which includes `std::to_string`. Use `format` instead.
```suggestion
				std::string default_town_number = fmt::format("{}", _settings_game.difficulty.number_towns == static_cast<uint>(CUSTOM_TOWN_NUMBER_DIFFICULTY) ? _settings_newgame.game_creation.custom_town_number : Map::ScaleBySize(_num_initial_towns[_settings_game.difficulty.number_towns]);
```